### PR TITLE
feat(ui5-switch): add required property

### PR DESCRIPTION
**Pull request by hinzzx**
Introducing the `required` property for the Switch control.
We now also enable the **Form Support** feature within the control, which means now the Switch component, could be easily used and integrated within forms as well.

Fixes: #5897

### CODE REVIEW

**Comment by nnaydenow** [comment]
Please also check what would be the bevahiour of the newly added property inside following scenario: https://github.com/SAP/ui5-webcomponents/issues/7319 (please rework it to work in your case by replacing ui5-checkbox with ui5-switch)

File: packages/main/src/Switch.ts [hunk 1]
```diff
@@ -188,8 +192,69 @@ class Switch extends UI5Element {
...
+		this._enableFormSupport();
+	}
+
+	_enableFormSupport() {
+		const formSupport = getFeature<typeof FormSupport>("FormSupport");
+		if (formSupport) {
+			formSupport.syncNativeHiddenInput(this, (element: IFormElement, nativeInput: HTMLInputElement) => {
+				const switchComponent = (element as Switch);
+				nativeInput.checked = !!switchComponent.checked;
+				nativeInput.disabled = !!switchComponent.disabled || switchComponent.checked;
```
Comment (nnaydenow) [inline packages/main/src/Switch.ts:251]: `nativeInput.disabled = !!switchComponent.disabled` seems to be enough for me

File: packages/main/test/pages/Switch.html [hunk 2]
```diff
@@ -42,6 +42,11 @@ <h3>Default Switch</h3>
 	</div>
 	<ui5-label id="lbl"></ui5-label>
 
+	<h3>Switch with required attribute</h3>
```
Comment (unazko) [inline packages/main/test/pages/Switch.html:45]: Could be "Required switch" and if not used for the tests I think we could only have an instance in form as sample.

File: packages/main/test/pages/Switch.html [hunk 3]
```diff
@@ -55,6 +60,15 @@ <h3>Graphical Switch</h3>
 		<ui5-switch design="Graphical" checked text-on="Yes" text-off="No" disabled></ui5-switch>
 	</div>
 
+	<h3>Switch in form, required</h3>
```
Comment (unazko) [inline packages/main/test/pages/Switch.html:63]: We could simulate a real use case. For example "Accept company policies" for a form label and "Yes"/"No" for the switch.

File: packages/playground/_stories/main/Switch/Switch.stories.ts [hunk 4]
```diff
@@ -60,4 +62,27 @@ export const Design = Template.bind({});
...
+	disabled: false,
+	design: SwitchDesign.Textual,
+};
+RequiredInForm.decorators = [
+	(story) => {
+		return html`
+		<form id="myForm">
+			<div style="width: fit-content; display: flex; flex-direction: column; justify-content: flex-start">
+				<div style="display: flex; flex-direction: column; justify-content: flex-start;">
+					<ui5-label for="mySwitch">The switch is required, and should be checked in order the form to be submit</ui5-label>
```
Comment (unazko) [inline packages/playground/_stories/main/Switch/Switch.stories.ts:80]: You could use something simpler as an explanation. Even better if we would simulate a real use case. Not sure how those stories are supposed to be written though.

**Review by unazko — COMMENTED** [review]
According to the screen reader announcements and attributes all seems to be working fine.
About the form support we I'll have to check.

File: packages/main/src/Switch.ts [hunk 5]
```diff
@@ -188,8 +192,70 @@ class Switch extends UI5Element {
...
+	}
+
+	_enableFormSupport() {
+		const formSupport = getFeature<typeof FormSupport>("FormSupport");
+		if (formSupport) {
+			formSupport.syncNativeHiddenInput(this, (element: IFormElement, nativeInput: HTMLInputElement) => {
+				const switchComponent = (element as Switch);
+				nativeInput.checked = !!switchComponent.checked;
+				nativeInput.disabled = !!switchComponent.disabled;
+				nativeInput.value = !!switchComponent.checked ? "on" : ""; // eslint-disable-line
```
Comment (nnaydenow) [inline packages/main/src/Switch.ts:252]: Do we need that eslint-disable here?

Comment (hinzzx) [reply packages/main/src/Switch.ts:252]: Actually, we could go without it. Thanks !

File: packages/main/src/Switch.ts [hunk 6]
```diff
@@ -188,8 +192,70 @@ class Switch extends UI5Element {
...
+	}
+
+	_enableFormSupport() {
+		const formSupport = getFeature<typeof FormSupport>("FormSupport");
+		if (formSupport) {
+			formSupport.syncNativeHiddenInput(this, (element: IFormElement, nativeInput: HTMLInputElement) => {
+				const switchComponent = (element as Switch);
+				nativeInput.checked = !!switchComponent.checked;
+				nativeInput.disabled = !!switchComponent.disabled;
+				nativeInput.value = switchComponent.checked ? "on" : "";
```
Comment (unazko) [inline packages/main/src/Switch.ts:252]: Maybe we should use switchComponent.textOn and switchComponent.textOff if such are present.

File: packages/playground/_stories/main/Switch/Switch.stories.ts [hunk 7]
```diff
@@ -60,4 +62,43 @@ export const Design = Template.bind({});
...
 	design: SwitchDesign.Graphical,
 	accessibleName: "graphical",
-};
\ No newline at end of file
+};
+
+export const RequiredInForm = Template.bind({});
+RequiredInForm.args = {
+	required: true,
+	name: "mySwitch",
```
Comment (unazko) [inline packages/playground/_stories/main/Switch/Switch.stories.ts:70]: "conditions" sound like a better name for the server to receive from the from submit.

File: packages/main/src/Switch.ts (hunk 6 above)
Comment (hinzzx) [reply packages/main/src/Switch.ts:252]: as discussed offline, this wont be necessary.

**Review by unazko — APPROVED** [review]
I'm fine with the change.

**Review by nnaydenow — APPROVED** [review]

**Review by nnaydenow — CHANGES_REQUESTED** [review]
Add test test validates the scenario in a form. You can use html like:

```html
	<form id="switchForm">
		<ui5-switch id="switch1" checked></ui5-switch>
		<ui5-switch id="switch2" text="Option 3" required></ui5-switch>
		<br><br>
		<ui5-button id="switchSubmit" type="Submit">Submit</ui5-button>
	</form>
```

And you can check how formElement.checkValidity() method works and maybe you can add it into the test

File: packages/main/test/specs/Switch.spec.js [hunk 8]
```diff
@@ -62,4 +62,29 @@ describe("Switch general interaction", async () => {
...
+		const requiredSwitch = await browser.$("#requiredTestSwitch");
+
+		let formValidity = await browser.execute(() => {
+			const form = document.getElementById("switchForm");
+			return form.checkValidity();
+		});
+
+		assert.strictEqual(formValidity, false, "The form could be submitted successfuly, when the 'required' switch is not checked");
+
+		requiredSwitch.setAttribute("checked", "");
```
Comment (nnaydenow) [inline packages/main/test/specs/Switch.spec.js:80]: requiredSwitch.click() should do the same

File: packages/main/test/pages/Switch.html [hunk 9]
```diff
@@ -83,6 +105,15 @@ <h3>sap_horizon</h3>
 <script>
 	var counter = 0;
 
+	submitBtn.addEventListener("click", function(e) {
+		if (myForm.checkValidity()) {
+			e.preventDefault();
+			alert("Form submitted successfully!");
+		} else {
```
Comment (nnaydenow) [inline packages/main/test/pages/Switch.html:112]: we dont need that empty else

**Review by nnaydenow — APPROVED** [review]

**Pull request merged**